In [2]:
import pandas as pd
airbnb = pd.read_csv('airbnb.csv')

## Using the method info()

Using the method info() we can obtain the answers for the following questions:

__Question 1__ - How many observations does the dataset airbnb have? <br>
__Question 2__ - Do we have missing values in the dataset airbnb?<br>
__Question 3__ - How many variables do not have any valid observations (all observations are missing)?<br>
__Question 4__ - How many integer variables do we have in our dataset?<br>

In [4]:
airbnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18723 entries, 0 to 18722
Data columns (total 20 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   room_id               18723 non-null  int64  
 1   survey_id             18723 non-null  int64  
 2   host_id               18723 non-null  int64  
 3   room_type             18723 non-null  object 
 4   country               0 non-null      float64
 5   city                  18723 non-null  object 
 6   borough               0 non-null      float64
 7   neighborhood          18723 non-null  object 
 8   reviews               18723 non-null  int64  
 9   overall_satisfaction  18723 non-null  float64
 10  accommodates          18723 non-null  int64  
 11  bedrooms              18723 non-null  float64
 12  bathrooms             0 non-null      float64
 13  price                 18723 non-null  float64
 14  minstay               0 non-null      float64
 15  name               

Answer to question... <br>
__1__ - 18723 Observations <br>
__2__ - Yes (Not all variables have 18273 non-null observations)<br>
__3__ - 4 (The variables with 0 non-null counting)<br>
__4__ - 5 variables (room_id, survey_id, host_id, reviews and accomodates)<br>

<hr>

## Using the method describe()

Using the method describe() we can obtain the answer for the following questions: <br>
__5__ - What is the maximum number of bedrooms available?<br>
__6__ - Half of the houses have a value lower than...<br>
__7__ - What is the mean value of overall_satisfaction (one decimal house)?<br>

In [3]:
airbnb.describe()

room_id  survey_id       host_id  country  borough       reviews  \
count  1.872300e+04    18723.0  1.872300e+04      0.0      0.0  18723.000000   
mean   1.120568e+07     1476.0  3.577612e+07      NaN      NaN     16.741548   
std    6.082192e+06        0.0  3.758103e+07      NaN      NaN     33.522630   
min    2.818000e+03     1476.0  2.234000e+03      NaN      NaN      0.000000   
25%    6.050608e+06     1476.0  7.140879e+06      NaN      NaN      2.000000   
50%    1.228287e+07     1476.0  1.988641e+07      NaN      NaN      6.000000   
75%    1.661084e+07     1476.0  5.202680e+07      NaN      NaN     17.000000   
max    2.000373e+07     1476.0  1.418319e+08      NaN      NaN    532.000000   

       overall_satisfaction  accommodates      bedrooms  bathrooms  \
count          18723.000000  18723.000000  18723.000000        0.0   
mean               3.301127      2.922021      1.430380        NaN   
std                2.213558      1.327524      0.879019        NaN   
min                0.000000      1.000000      0.000000        NaN   
25%                0.000000      2.000000      1.000000        NaN   
50%                4.500000      2.000000      1.000000        NaN   
75%                5.000000      4.000000      2.000000        NaN   
max                5.000000     17.000000     10.000000        NaN   

              price  minstay      latitude     longitude  
count  18723.000000      0.0  18723.000000  18723.000000  
mean     166.599477      NaN     52.365261      4.888585  
std      108.943849      NaN      0.015142      0.034537  
min       12.000000      NaN     52.296200      4.763264  
25%      108.000000      NaN     52.355254      4.864344  
50%      144.000000      NaN     52.364628      4.885994  
75%      192.000000      NaN     52.374797      4.907480  
max     6000.000000      NaN     52.424980      5.027689

Answer to question... <br>
__5__ - 10 bedrooms (Check the max) <br>
__6__ - 144 (Check the median)<br>
__7__ - 3.3 (Check the mean)<br>
<hr>

Using the method describe(include = ['O']) we can obtain the answer for the following questions: <br>
__8__ - The most frequent neighborhood is...<br>
__9__ - The most frequent type in 'room_type' is...<br>

In [4]:
airbnb.describe(include = ['O'])

room_type       city            neighborhood       name  \
count             18723      18723                   18723      18671   
unique                3          1                      23      18150   
top     Entire home/apt  Amsterdam  De Baarsjes / Oud West  Amsterdam   
freq              14978      18723                    3289         36   

                     last_modified  \
count                        18723   
unique                       18723   
top     2017-07-22 17:27:56.466332   
freq                             1   

                                                 location  
count                                               18723  
unique                                              18723  
top     0101000020E61000000133DFC14F7C1340B459F5B9DA30...  
freq                                                    1

Answer to question... <br>
__8__ - De Baarsjes / Oud West (Check the top) <br>
__9__ - Entire home/apt (Check the top)<br>
<hr>

## Using the method value_counts()

Using the method value_counts() we can obtain the answer for the following questions: <br>
__10__ - How many houses can accommodate more than 15 people? <br>
__11__ - How many houses accommodates only one person?

In [5]:
airbnb['accommodates'].value_counts()

2     10024
4      5579
3      1585
6       476
5       471
1       367
8       105
7        52
16       20
10       16
12       10
9         8
14        6
11        2
13        1
17        1
Name: accommodates, dtype: int64

Answer to question... <br>
__10__ - 21 houses (sum of houses with 16 and 17 in accomodates) <br>
__11__ - 367 houses <br>
<hr>

## Using the method groupby()

Using the method value_counts() we can obtain the answer for the following questions: <br>
__12__ - What is the mean price for houses with an overall_satisfation of 5.0 (1 decimal houses)? <br>
__13__ - What is the mean price of a house that hosts 16 people?

In [6]:
airbnb.groupby('overall_satisfaction')['price'].mean()

overall_satisfaction
0.0    175.313848
1.0     60.000000
1.5     25.000000
2.5    720.000000
3.0    151.052632
3.5    142.908257
4.0    143.942808
4.5    152.902610
5.0    170.231967
Name: price, dtype: float64

In [7]:
airbnb.groupby('accommodates')['price'].mean()

accommodates
1      79.201635
2     131.714086
3     151.516088
4     210.716257
5     240.959660
6     302.983193
7     314.096154
8     407.142857
9     412.000000
10    552.437500
11    507.000000
12    547.200000
13    318.000000
14    511.000000
16    745.700000
17    420.000000
Name: price, dtype: float64

Answer to question... <br>
__12__ - 170.2 <br>
__13__ - 745.7 <br>
<hr>

## Using the method corr()

Using the method corr() we can obtain the answer for the following questions: <br>
__14__ - What is the variable that has the higher correlation (Pearson) with the target (price)? <br>
__15__ - What is the correlation (pearson) between reviews and overall_satisfaction (2 decimal houses)? <br>

In [8]:
airbnb.corr(method = 'pearson')

room_id  survey_id   host_id  country  borough  \
room_id               1.000000        NaN  0.535106      NaN      NaN   
survey_id                  NaN        NaN       NaN      NaN      NaN   
host_id               0.535106        NaN  1.000000      NaN      NaN   
country                    NaN        NaN       NaN      NaN      NaN   
borough                    NaN        NaN       NaN      NaN      NaN   
reviews              -0.390703        NaN -0.183331      NaN      NaN   
overall_satisfaction -0.386261        NaN -0.203703      NaN      NaN   
accommodates         -0.004256        NaN  0.014030      NaN      NaN   
bedrooms             -0.029269        NaN -0.016995      NaN      NaN   
bathrooms                  NaN        NaN       NaN      NaN      NaN   
price                -0.008476        NaN -0.016390      NaN      NaN   
minstay                    NaN        NaN       NaN      NaN      NaN   
latitude             -0.033042        NaN -0.026079      NaN      NaN   
longitude             0.010033        NaN  0.010722      NaN      NaN   

                       reviews  overall_satisfaction  accommodates  bedrooms  \
room_id              -0.390703             -0.386261     -0.004256 -0.029269   
survey_id                  NaN                   NaN           NaN       NaN   
host_id              -0.183331             -0.203703      0.014030 -0.016995   
country                    NaN                   NaN           NaN       NaN   
borough                    NaN                   NaN           NaN       NaN   
reviews               1.000000              0.316705     -0.061360 -0.114641   
overall_satisfaction  0.316705              1.000000     -0.053410 -0.087664   
accommodates         -0.061360             -0.053410      1.000000  0.711051   
bedrooms             -0.114641             -0.087664      0.711051  1.000000   
bathrooms                  NaN                   NaN           NaN       NaN   
price                -0.084516             -0.044775      0.500350  0.446144   
minstay                    NaN                   NaN           NaN       NaN   
latitude              0.050451              0.041197      0.003836 -0.011094   
longitude            -0.012108             -0.017531      0.104110  0.095636   

                      bathrooms     price  minstay  latitude  longitude  
room_id                     NaN -0.008476      NaN -0.033042   0.010033  
survey_id                   NaN       NaN      NaN       NaN        NaN  
host_id                     NaN -0.016390      NaN -0.026079   0.010722  
country                     NaN       NaN      NaN       NaN        NaN  
borough                     NaN       NaN      NaN       NaN        NaN  
reviews                     NaN -0.084516      NaN  0.050451  -0.012108  
overall_satisfaction        NaN -0.044775      NaN  0.041197  -0.017531  
accommodates                NaN  0.500350      NaN  0.003836   0.104110  
bedrooms                    NaN  0.446144      NaN -0.011094   0.095636  
bathrooms                   NaN       NaN      NaN       NaN        NaN  
price                       NaN  1.000000      NaN  0.013759   0.021770  
minstay                     NaN       NaN      NaN       NaN        NaN  
latitude                    NaN  0.013759      NaN  1.000000  -0.107549  
longitude                   NaN  0.021770      NaN -0.107549   1.000000

Answer to question... <br>
__14__ - accomodates (0.5) <br>
__15__ - 0.32 <br>
<hr>